### ISSUE TITLE
Support for *custom* authorizers in API Gateway methods 

### ISSUE THREAD

**Issue by deontaljaard** [localstack/localstack#1315]
Does localstack's mock implementation of API Gateway support custom authorizers? Potentially related - [#10180](https://github.com/localstack/localstack/issues/1080). Using localstack-0.8.7.

My reason for asking is, when I issue a put-method with an authorizer-id, the authorizer-id is left null after the method's creation. Also, there's a difference in the response payload between localstack and AWS when issuing a similar request.

### put-method request against localstack
```bash
awslocal apigateway put-method \
  --rest-api-id $rest_api_id \
  --resource-id $resource_id \
  --http-method GET\
  --authorization-type "CUSTOM" \
  --authorizer-id $authorizer_id \
  --api-key-required \
  --profile local-cog
```

### method description @ http://localhost:4567/restapis/13A-Z1686799/resources/779367844A-Z
```json
{"path": "/engine/api/products", "parentId": "079240A-Z373", "id": "779367844A-Z", "pathPart": "engine/api/products", "resourceMethods": {"GET": {"apiKeyRequired": null, "httpMethod": "GET", "methodIntegration": null, "requestParameters": null, "requestModels": null, "authorizationType": "CUSTOM", "authorizerId": null}}}
```

I can see the authorization type of the method as _CUSTOM_, however the authorizerId is _null_. (Note apiKeyRequired is _null_ too)

### put-method response against localstack
```json
{
    "authorizationType": "CUSTOM",
    "httpMethod": "GET"
}
```

No authorizerId in response. (Note no apiKeyRequired field too)

### put-method request against aws
```bash
aws apigateway put-method \
  --rest-api-id $rest_api_id \
  --resource-id $resource_id \
  --http-method GET \
  --authorization-type "CUSTOM" \
  --authorizer-id $authorizer_id \
  --api-key-required \
  --profile qa-cog
```

### put-method response against aws
```json
{
    "httpMethod": "GET",
    "authorizerId": "84241l",
    "apiKeyRequired": false,
    "authorizationType": "CUSTOM"
}
```

The response contains the authorizerId and apiKeyRequired. 

Any guidance and/or assistance will be appreciated.

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-99) by [Unito](https://www.unito.io/learn-more)

**Comment by pmcdowell-okta** [comment]
I too am experimenting with API Gateway and Custom Authorizers, they seem to be ignored in Localstack unfortunately.

**Comment by jwcnewton** [comment]
Any movement on this?

**Comment by PavelShishmarev** [comment]
Is there are any updates?

**Comment by monty16597** [comment]
Hello, @deontaljaard @pmcdowell-okta @jwcnewton @PavelShishmarev
@deontaljaard Thanks for raising the issue and sorry that we are this late to give an update on this. 
You are right that the response is diverging from the response of the AWS for the `put_method` of the API gateway. We have merged the fix already (#3155). However, invoking the custom authorizer is being supported in the PRO version of the localstack only.
We have also integrated some test [here](https://github.com/localstack/localstack/blob/cba8fdeeeed0c1621775297e89b7c1f3696a11ae/tests/integration/test_api_gateway.py#L493-L529).
